File: advance/src/eu/advance/logistics/flow/model/AdvanceCompositeBlockParameterDescription.java
/*
 * Copyright 2010-2012 The Advance EU 7th Framework project consortium
 *
 * This file is part of Advance.
 *
 * Advance is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Advance is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with Advance.  If not, see
 * <http://www.gnu.org/licenses/>.
 *
 */

package eu.advance.logistics.flow.model;

import java.util.List;

import com.google.common.collect.Lists;

import edu.umd.cs.findbugs.annotations.Nullable;
import eu.advance.logistics.util.Strings;
import eu.advance.logistics.xml.typesystem.XElement;

/**
 * The input or output parameter description of a flow composite block element.
 * @author karnokd, 2011.06.21.
 */
public class AdvanceCompositeBlockParameterDescription extends
		AdvanceBlockParameterDescription {
	/** The user-entered documentation of this parameter. */
	@Nullable
	public String documentation;
	/** The user-entered keywords for easier finding of this parameter. */
	public final List<String> keywords = Lists.newArrayList();
	
	/**
	 * Load a parameter description from an XML element which conforms the {@code flow-description.xsd}.
	 * @param root the element of a input or output
	 */
	@Override
	public void load(XElement root) {
		super.load(root);
		documentation = root.get("documentation");
		String kw = root.get("keywords");
		if (kw != null) {
			keywords.addAll(Strings.trim(Strings.split(kw, ',')));
		}
	}
}

File: advance-api-cs/AdvanceAPIClient/Classes/Model/AdvanceCompositeBlockParameterDescription.cs
/*
 * Copyright 2010-2013 The Advance EU 7th Framework project consortium
 *
 * This file is part of Advance.
 *
 * Advance is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Advance is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with Advance.  If not, see
 * <http://www.gnu.org/licenses/>.
 *
 */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Globalization;
using System.Xml;

using AdvanceAPIClient.Core;

namespace AdvanceAPIClient.Classes.Model
{
    /// <summary>
    /// Input or output parameter description of a flow composite block element.
    /// </summary>
    public class AdvanceCompositeBlockParameterDescription : AdvanceBlockParameterDescription
    {
        /// <summary>
        /// User-entered keywords for easier finding of this parameter.
        /// </summary>
        public List<string> Keywords = new List<string>();

        /// <summary>
        /// Fill object data from xml node
        /// </summary>
        /// <param name="source"></param>
        protected override void LoadFromXmlNode(XmlNode source)
        {
            base.LoadFromXmlNode(source);
            this.Keywords = GetListAttribute(source, "keywords");
        }
        /// <summary>
        /// Fills object's xml
        /// </summary>
        /// <param name="node">Empty node for the object with name "tagname"</param>
        protected override void FillXmlElement(XmlElement node)
        {
            base.FillXmlElement(node);
            AddListAttribute(node, "keywords", this.Keywords);
        }
    }
}
